### ISSUE TITLE
net/frr: FRR bgpd prefix list WebGUI does not check prefix list name and will lead to generating wrong config files.

### ISSUE THREAD

**Issue by SuperMarioSF** [opnsense/plugins#1407]
Issue updated

# Details

I have found the old issue is part of my fault. _(Old issue listed below)_
But there does have no check about the spaces in prefix name on bgpd prefix-list configuration. This will lead to generate wrong configuration file, if user typed in a prefix list name with a space character between words.

## The exact URL of the GUI page involved
`https://your.opnsense.instance/ui/quagga/bgp/index` at 'Prefix list' tab

## A list of steps to replicate the bug
1. Install and enable FRR
2. Set some prefix list **with prefix list name contained space between words**
3. Add some neighbor
4. Enable `bgpd`
5. Go to SSH and use `vtysh` to show current effective config (`show running-config`)
6. View `/usr/local/etc/frr/bgpd.conf` to compare difference

## Last known versions of this issue
Same as old issue.

* * *

# Old issue

## TL;DR
Something wrong with FRR plugin. Configuration generator did not generated correct configuration.

## Details
I have defined multiple prefix list from WebGUI buy none of them worked.  
I opened configuration file of bgpd and I saw there is configuration like this:
```
ip prefix-list SomeName routes seq 10 permit 192.168.1.0/24
```
I tried enter `vtysh` and configure like this but got denied by `Unknown command`.
I tried and found the **correct** syntax for that prefix:
```
ip prefix-list SomeName seq 10 permit 192.168.1.0/24
```

You see there should not have `routes` there. May be due to FRR's update.

## Last known versions of this issue
I haven't test other version, but current listed below is affected for sure:
```
FRR bgpd version 6.0.2
OPNsense 19.1.10-amd64 stable (5e5adf26fc3)
OpenSSL 1.0.2s

Package 'frr6' version 6.0.2_2
Package 'os-frr' version 1.11
```

## The exact URL of the GUI page involved
`https://your.opnsense.instance/ui/quagga/bgp/index` at 'Prefix list' tab

## A list of steps to replicate the bug
1. Install and enable FRR
2. Set some prefix list
3. Add some neighbor
4. Enable `bgpd`
5. Go to SSH and use `vtysh` to show current effective config (`show running-config`)
6. View `/usr/local/etc/frr/bgpd.conf` to compare difference

**Comment by fichtner** [comment]
Same as https://github.com/opnsense/plugins/issues/1400 and https://github.com/opnsense/plugins/issues/1401 ?

**Comment by mimugmail** [comment]
I really have no idea where this "routes" thing come from.
Did you set a space in prefix-list name? On my side with latest master it works fine.

**Comment by SuperMarioSF** [comment]
> 
> 
> I really have no idea where this "routes" thing come from.
> Did you set a space in prefix-list name? On my side with latest master it works fine.

Hmmmm maybe you found the exact problem.
It sure that I have space in prefix-list name and in my configuration it does have a "routes" in prefix name. That is my fault.
But I wonder why there did not have any check for this condition.
I will change the issue title and description for this unchecked condition.

**Comment by SuperMarioSF** [comment]
> 
> 
> Same as #1400 and #1401 ?

Not exactly. This problem should caused by a unchecked user input.

**Comment by mimugmail** [comment]
Ok, this shouldn't be hard to add. Thx for testing/reporting!
